src/packet_analysis/Analyzer.cc

```diff
@@ -204,4 +241,16 @@ void Analyzer::AnalyzerViolation(const char* reason, session::Session* session,
 	event_mgr.Enqueue(analyzer_violation, session->GetVal(), tval, val_mgr->Count(0), std::move(r));
 	}
 
+void Analyzer::AnalyzerViolation(const char* reason, session::Session* session, const char* data,
+                                 int len, zeek::Tag arg_tag)
+	{
+	session->SetAnalyzerState(GetAnalyzerTag(), session::AnalyzerConfirmationState::VIOLATED);
+
+	if ( analyzer_violation_info )
+		EnqueueAnalyzerViolationInfo(session, reason, data, len, tag);
+
+	if ( analyzer_violation )
+		EnqueueAnalyzerViolation(session, reason, data, len, tag);
```

Comment:
This could likewise get expensive if the same violation keeps triggering for many packets. Not sure what to do about that ... (could do some thresholding/limiting, but not straight-forward how).

Reply:
Yes. See reply above.